lib/render/OSPRay.cpp

```diff
@@ -0,0 +1,201 @@
+#include <vapor/OSPRay.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+using namespace VOSP;
+
+#ifdef BUILD_OSPRAY
+static int _initialized = false;
+
+void ospErrorCallback(OSPError error, const char *msg)
+{
+    fprintf(stderr, "OSP error ");
+    switch (error) {
+#define STRINGIFY(x) case x: fprintf(stderr, #x); break;
+        STRINGIFY(OSP_NO_ERROR)
+        STRINGIFY(OSP_UNKNOWN_ERROR)
+        STRINGIFY(OSP_INVALID_ARGUMENT)
+        STRINGIFY(OSP_INVALID_OPERATION)
+        STRINGIFY(OSP_OUT_OF_MEMORY)
+        STRINGIFY(OSP_UNSUPPORTED_CPU)
+        STRINGIFY(OSP_VERSION_MISMATCH)
+#undef STRINGIFY
+    }
+    fprintf(stderr, ": %s\n", msg);
+//    exit(1);
+}
+#endif
+
+int VOSP::Initialize(int *argc, char **argv)
+{
+#ifdef BUILD_OSPRAY
+    if (_initialized)
+        return 0;
+    
+    OSPError init_error = ospInit(argc, (const char **)argv);
+    if (init_error != OSP_NO_ERROR)
+      return init_error;
+    
+// Hide deprecated warning.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+    // If this is not set, OSPRay will crash upon shutdown
+    ospDeviceSetErrorFunc(ospGetCurrentDevice(), ospErrorCallback);
+#pragma GCC diagnostic pop
+    
+    printf("OSPRay Version = %s\n", Version().c_str());
```

Comment:
Remove/disable diagnostics.

Reply:
Done

Reply:
(I currently need this for testing so I added it back today after removing it yesterday. I moved it so it won't print anything)